### ISSUE TITLE
Change runserver to Gunicorn

### ISSUE THREAD

**Issue by Braintelligence** [galexrt/docker-healthchecks#4]
Hi, I'm trying to run this within Rancher by using https://github.com/adi90x/rancher-active-proxy

I get these errors:
```
27.12.2017 13:59:30nginx.1     | 2017/12/27 12:59:30 [error] 2680#2680: *1726 connect() failed (111: Connection refused) while connecting to upstream, client: client.ip.add.ress, server: my.domain.com, request: "GET / HTTP/2.0", upstream: "http://10.42.29.44:80/", host: "my.domain.com"

27.12.2017 13:59:30nginx.1     | 2017/12/27 12:59:30 [error] 2680#2680: *1726 no live upstreams while connecting to upstream, client: client.ip.add.ress, server: my.domain.com, request: "GET / HTTP/2.0", upstream: "http://my.domain.com/", host: "my.domain.com"
```

Maybe using "0.0.0.0:8000" for runserver is the culprit?

I connect a random local host port to the container port 8000, then rancher-active-proxy works with that. Works flawlessly for many containers up till now.

**Comment by galexrt** [comment]
Could you exec into the container and run `ss -tulpn`. The output should look like this:
```
# ss -tulpn
Netid  State      Recv-Q Send-Q     Local Address:Port       Peer Address:Port 
tcp    LISTEN     0      10                     *:8000                  *:*
```

**Comment by Braintelligence** [comment]
![grafik](https://user-images.githubusercontent.com/6761196/34383444-7e1409a2-eb15-11e7-9073-7c74a45dafce.png)

**Comment by Braintelligence** [comment]
Ok sorry, I'm dumb. I set the wrong port in the labels that the proxy container is listening to. It tried to connect to port 80 on the healthchecks container.

How I get this as DJANGO error, though:
DisallowedHost at /
Invalid HTTP_HOST header: 'my.domain.com'. You may need to add 'my.domain.com' to ALLOWED_HOSTS.

I know I can set this in settings.py (I'm a Django user myself) but maybe there's an environment variable for this?

Also: I thought it is bad to use the development runserver for production? Now people can see debug messages =(.

**Comment by galexrt** [comment]
About `ALLOWED_HOSTS` you would add an env var named `HC_ALLOWED_HOSTS` with value `["my.domain.com"]` for example (yes with the square brackets and quotes).

I'm not a python developer, I just followed their instructions so I could run healthchecks.
If you let me know what server/application is normally used to run healthchecks in production I'm happy to implement it or take a PR. :)

**Comment by Braintelligence** [comment]
Oh sure, normally you use Gunicorn for the WSGI interface and just let it run at startup.

Here's a documentation on how to do this:
- https://www.digitalocean.com/community/tutorials/how-to-set-up-django-with-postgres-nginx-and-gunicorn-on-debian-8#adjust-the-project-settings
- https://www.digitalocean.com/community/tutorials/how-to-set-up-django-with-postgres-nginx-and-gunicorn-on-debian-8#create-a-gunicorn-systemd-service-file

Since we don't need a virtual environment you can just go on and install gunicorn here https://github.com/galexrt/docker-healthchecks/blob/master/Dockerfile#L15

For ease of use you should bind Gunicorn to a port, and not a unix socket, as is described here: https://www.digitalocean.com/community/tutorials/how-to-set-up-django-with-postgres-nginx-and-gunicorn-on-debian-8#testing-gunicorn-39-s-ability-to-serve-the-project

So basically you need to:
1. Install Gunicorn when building the image
2. Create a Gunicorn startup service, where you bind Gunicorn to 0.0.0.0:8000 and point it to the WSGI interface of the Django project 

I don't have time for a PR currently, but if you run into problems I'd be glad to help!

**Comment by galexrt** [comment]
Thanks! I'll look into it. :+1: 
I'll try to get to it on the weekend, when I have the time. :)

**Comment by Braintelligence** [comment]
Thanks, that would be awesome. Until then I'll be using this for a test-drive.

**Comment by Braintelligence** [comment]
Is the current commit working? It says `ImportError: No module named django.core.management` when trying to run, which normally means, that Django isn't installed for the invoking python binary.

**Comment by galexrt** [comment]
The issues are now fixed. At least for me the basic web interface worked, couldn't test login because SSL is required for the CSRF tokens..
It now doesn't use virtual env, but now uses gunicorn and nginx for production usability.

**Comment by Braintelligence** [comment]
I'm building it right now.
I create the images by using the docker-compose build from git route.

**Comment by Braintelligence** [comment]
One thing that pops to my mind is that you forgot to set `DEBUG = False`, but I can't get it to run currently. I get HTTP 500.
Trying to get it to work now.

**Comment by Braintelligence** [comment]
Something seems to have went wrong when installing build-essentials and python3-dev, because I couldn't even run python3 manage.py runserver within the container. Django was missing.
Trying to pip install django didn't work because build tools were missing.

Is this because you ran pip install with `--user` ?

**Comment by Braintelligence** [comment]
I see, you're removing everything after install.
Well it's getting really late. I will look into this tomorrow.
But for now I can't get this to work =/. I only get HTTP 500 and just can't tell what's going on, because the logs aren't showing anything relevant to me.

**Comment by galexrt** [comment]
The issue was that a call to `manage.py compress` is needed when running `DEBUG = False`.
This has been fixed now in the latest commits.
Now I'm getting a HTTP 200 and a working Healthchecks frontend.

**Comment by Braintelligence** [comment]
Yeah it works now. Awesome. I guess this can be used productively now 👍 (I do, at least)

**Comment by galexrt** [comment]
@Braintelligence I hope so. :-) Could you please go ahead and check the issues you created if they have been resolved or still need work done?
If you have any other questions or find something that isn't documented well enough, please let me know.

**Comment by Braintelligence** [comment]
Well this is resolved
